pkg/model/components/docker.go

```diff
@@ -50,40 +45,33 @@ func (b *DockerOptionsBuilder) BuildOptions(o interface{}) error {
 	// Container runtime is not Docker, should not install
 	if clusterSpec.ContainerRuntime != "docker" {
 		docker.SkipInstall = true
+		return nil
 	}
 
+	// Set the Docker version for known Kubernetes versions
 	if fi.StringValue(clusterSpec.Docker.Version) == "" {
-		if clusterSpec.KubernetesVersion == "" {
-			return fmt.Errorf("KubernetesVersion is required")
-		}
-
-		dockerVersion := ""
-		if sv.Major == 1 && sv.Minor >= 17 {
-			dockerVersion = "19.03.4"
-		} else if sv.Major == 1 && sv.Minor >= 16 {
-			dockerVersion = "18.09.9"
-		} else if sv.Major == 1 && sv.Minor >= 12 {
-			dockerVersion = "18.06.3"
-		} else if sv.Major == 1 && sv.Minor >= 9 {
-			dockerVersion = "17.03.2"
-		} else if sv.Major == 1 && sv.Minor >= 8 {
-			dockerVersion = "1.13.1"
-		} else if sv.Major == 1 && sv.Minor >= 6 {
-			dockerVersion = "1.12.6"
-		} else if sv.Major == 1 && sv.Minor >= 5 {
-			dockerVersion = "1.12.3"
-		} else if sv.Major == 1 && sv.Minor <= 4 {
-			dockerVersion = "1.11.2"
+		if b.IsKubernetesGTE("1.17") {
+			docker.Version = fi.String("19.03.4")
+		} else if b.IsKubernetesGTE("1.16") {
+			docker.Version = fi.String("18.09.9")
+		} else if b.IsKubernetesGTE("1.12") {
+			docker.Version = fi.String("18.06.3")
+		} else if b.IsKubernetesGTE("1.9") {
+			docker.Version = fi.String("17.03.2")
+		} else if b.IsKubernetesGTE("1.8") {
+			docker.Version = fi.String("1.13.1")
+		} else if b.IsKubernetesGTE("1.6") {
+			docker.Version = fi.String("1.12.6")
+		} else if b.IsKubernetesGTE("1.5") {
+			docker.Version = fi.String("1.12.3")
+		} else if b.IsKubernetesGTE("1.4") {
+			docker.Version = fi.String("1.11.2")
+		} else {
```

Comment:
This is a semantic change. Previously it would pick 1.11.2.

Reply:
True. I think at that time I very tempted to remove the defaults these completely as they will be removed in 1.18.
I guess will do it in a future commit. Thanks,